test/com/google/javascript/jscomp/parsing/NewParserTest.java

```diff
@@ -2648,9 +2648,34 @@ public void testImport() {
     parse("import {x, y} from './someModule'");
     parse("import {x as x1, y as y1} from './someModule'");
     parse("import {x as x1, y as y1, } from './someModule'");
+    parse("import {default as d} from './someModule'");
+    parse("import d, {x as x1, y as y1} from './someModule'");
     parse("import * as sm from './someModule'");
   }
 
+  public void testExport() {
+    mode = LanguageMode.ECMASCRIPT6;
+
+    parse("export const x = 1");
+    parse("export var x = 1");
+    parse("export function f() {}");
+    parse("export class c {}");
+    parse("export {x, y}");
+    parse("export {x as x1}");
+    parse("export {x as default, y as y1}");
+    parseError("export {default as x}",
+        "cannot use keyword 'default' here.");
+    parseError("export {package as x}",
+        "cannot use keyword 'package' here.");
+
+    parse("export {x as x1, y as y1} from './someModule'");
+    parse("export {x as x1, y as y1, } from './someModule'");
+    parse("export {default as d} from './someModule'");
+    parse("export {x as default, y as y1} from './someModule'");
+    parse("export {default as default, y as y1} from './someModule'");
+    parse("export * from './someModule'");
+  }
+
   public void testShebang() {
```

Comment:
Can we add some more tests of the error cases?
In particular, I think we need:

```javascript
import keyword from '...';
import {nonkeyword as keyword} from '...';
import {keyword} from '...';
export {keyword};````